doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1161 @@
+1                  DESIGNATED   FORWARDING      1000        128.45       2, 300, 400
+```
+
+
+- show spanning_tree bpdu_guard
+
+```
+admin@sonic:  show spanning_tree bpdu_guard
+
+
+PortNum               Shutdown Configured             Port Shut due to BPDU Guard
+---------             --------------------           -----------------------------
+Ethernet46             Yes                            Yes             
+PortChannel1001        No                             NA
+```
+
+- show spanning_tree  mst instance \<instance-id\> interface \<ifname\>
+
+```
+admin@sonic:  show spanning_tree  mst instance 0 interface Ethernet46
+
+
+Port        Prio   Path    Edge     State         Role     Designated    Designated          Designated
+Num         rity   Cost    Port                            Cost          Root                Bridge
+Ethernet46  128    800     N        FORWARDING    Root     0             32768002438eefbc3   32768002438eefbc3
+```
+
+### Statistics Commands
+- show spanning_tree mst statistics instance \<instance-id\>
+
+```
+
+admin@sonic:  show spanning_tree  mst  statistics instance 0 
+
+
+MSTP instance 0 - VLANs 1, 4-8, 202-4094
+--------------------------------------------------------------------
+PortNum           BPDU Tx     BPDU Rx     TCN Tx     TCN Rx             
+Ethernet20        80	      4           3          4
+Ethernet46        40	      6           3          4
+PortChannel15     30	      6           4          1
+```
+
+
+## Clear Commands
+- sonic-clear spanning_tree statistics 
+- sonic-clear spanning_tree mst statistics instance \<instance-id\>
+
+
+## Debug Commands
+Following debug commands will be supported for enabling additional logging which can be viewed in /var/log/stpd.log, orchagent related logs can be viewed in /var/log/syslog.
+
+- debug spanning_tree mst instance \<instance-id\>
+- debug spanning_tree bpdu [tx|rx]
+- debug spanning_tree event
+- debug spanning_tree verbose
+- debug spanning_tree interface \<ifname\>
+
+Following debug commands will be supported for displaying internal data structures
+
+- debug spanning_tree dump global 
+- debug spanning_tree dump mst instance \<instance-id\>
+
+# YANG Model
+
+YANG Model will be extended as follows for MSTP:
+```yang
+module sonic-stp {
+
+    yang-version 1.1;
+
+    namespace "http://github.com/sonic-net/sonic-stp";
+    prefix "stp";
+
+    import sonic-port {
+        prefix port;
+        revision-date 2019-07-01;
+    }
+
+    import sonic-portchannel {
+        prefix lag;
+        revision-date 2021-06-13;
+    }
+
+    import sonic-vlan {
+        prefix vlan;
+        revision-date 2021-04-22;
+    }
+
+    import sonic-device_metadata { 
+        prefix device_metadata;
+        revision-date 2021-02-27;
+    }
+
+    description "STP yang Module for SONiC OS";
+
+    revision 2023-04-18 {
+        description "First Revision";
+    }
+
+    grouping interfaceAttr {
```

Comment:
Should this interface grouping also have leaf node for link-type setting? Kindly check.

Reply:
Updated in HLD as per suggestion